src/Processors/QueryPlan/MergingAggregatedStep.cpp

```diff
@@ -127,15 +137,19 @@ void MergingAggregatedStep::transformPipeline(QueryPipelineBuilder & pipeline, c
         pipeline.resize(1);
 
         /// Now merge the aggregated blocks
-        pipeline.addSimpleTransform([&](const Block & header)
-                                    { return std::make_shared<MergingAggregatedTransform>(header, transform_params, max_threads); });
+        auto transform = std::make_shared<MergingAggregatedTransform>(pipeline.getHeader(), params, final, grouping_sets_params, max_threads);
+        pipeline.addTransform(std::move(transform));
     }
     else
     {
+        if (input_streams.front().header.has("__grouping_set") || !grouping_sets_params.empty())
+            throw Exception(ErrorCodes::LOGICAL_ERROR,
+                 "Memory efficient merging of aggregated results is not supported for grouping sets.");
```

Comment:
I'm just curious why?

Reply:
Because nobody has implemented it.
It is disabled with grouping sets, and I do not want to support it now.